File: proposals/PROP-001/antennas/antenna_lookup.py
"""
antenna_lookup.py  --  Builds and exposes the runtime lookup dict.
Import this anywhere you need antenna resolution.
"""
# Copyright (C) 2026 Open HamClock Backend (OHB) Contributors and David Strickland KR8X
# License: GNU Affero General Public License v3.0 (AGPLv3)
# See LICENSE file or <https://www.gnu.org/licenses/agpl-3.0.html>
from antenna_data import ANTENNA_DATA

# key = msb*256 + lsb  →  {'path': ..., 'description': ...}
antenna_lookup: dict[int, dict[str, str]] = {
    msb * 256 + lsb: {'path': path, 'description': desc}
    for msb, lsb, path, desc in ANTENNA_DATA
}


def lookup_antenna(index: int) -> dict[str, str] | None:
    """Return {'path', 'description'} for a given msb*256+lsb index, or None."""
    return antenna_lookup.get(index)

File: proposals/PROP-001/antennas/antenna_lookup.cpp
// antenna_lookup.cpp
// Copyright(c) 2026 HamClock Contributors and David Strickland KR8X
// Licensed under the MIT License.
// See LICENSE file in the project root for full license information
#include "antenna_lookup.h"
#include "antenna_data.h"       // generated -- contains ANTENNA_DATA[]

#include <unordered_map>
#include <cstdint>

static std::unordered_map<uint16_t, AntennaInfo> s_table;

void antenna_lookup_init()
{
	s_table.reserve(ANTENNA_DATA_COUNT);
	for (std::size_t i = 0; i < ANTENNA_DATA_COUNT; ++i) {
		const AntennaEntry& e = ANTENNA_DATA[i];
		s_table[e.index] = AntennaInfo{ e.path, e.description };
	}
}

const AntennaInfo* antenna_lookup(uint16_t index)
{
	auto it = s_table.find(index);
	return (it != s_table.end()) ? &it->second : nullptr;
}